### ISSUE TITLE
grails.tomcat.certificateFile and certificateKeyFile ignored when running grails run-app -https

### ISSUE THREAD

**Issue by edwardotis** [grails-plugins/grails-tomcat-plugin#16]
build ":tomcat:8.0.30"
grails 2.5.4
Java 8.77
Ubuntu 14.04 LTS

I configured Buildconfig.groovy to point to my own openssl self signed cert and key within my directory structure.
grails.tomcat.certificateFile="test/test_data/self_signed_ssl_cert_for_dev/development.crt"
grails.tomcat.certificateKeyFile="test/test_data/self_signed_ssl_cert_for_dev/development.key"

However, grails creates a different cert to use. It doesn't use the configured one with the tomcat plugin.

./grailsw run-app -https
Creating SSL Certificate...
Created SSL Certificate.

This causes my selenium tests to fail on my continuous integration server. I need to provide my own credentials.
I've used this crt and key in gunicorn before, and need to use them in this project as well.
(They are only used for development and continuous integration, not production.)

Ideas?

**Comment by edwardotis** [comment]
I see that it's supported in the master branch:
org.grails.plugins.tomcat.TomcatServer#configureSsl

```
def certificateKeyFile = getConfigParam('certificateKeyFile') ?: ''
		def certificateFile = getConfigParam('certificateFile') ?: ''
		if (certificateKeyFile && certificateFile) {
			sslConnector.setAttribute 'SSLHonorCipherOrder', false
			sslConnector.setAttribute 'SSLCertificateKeyFile', certificateKeyFile
			sslConnector.setAttribute 'SSLCertificateFile', certificateFile
			def certificateKeyPassword = getConfigParam('certificateKeyPassword') ?: ''
			if (certificateKeyPassword) {
				sslConnector.setAttribute 'SSLPassword', certificateKeyPassword
			}
		}
		else {
			sslConnector.setAttribute 'keystoreFile', keystoreFile.absolutePath
			sslConnector.setAttribute 'keystorePass', keyPassword

			if (truststoreFile.exists()) {
				CONSOLE.addStatus "Using truststore $truststore"
				sslConnector.setAttribute 'truststoreFile', truststore
				sslConnector.setAttribute 'truststorePass', trustPassword
				sslConnector.setAttribute 'clientAuth', getConfigParam('clientAuth') ?: 'want'
			}
		}

```

But support was dropped in the 7.x and 8.x branches. Those skip the certifcateFile and certificateKeyFile option and force the keystoreFile and keystorePass option.

certificateFile and certificateKeyFile are supported in Tomcat 8:
https://tomcat.apache.org/tomcat-8.5-doc/config/http.html#SSL_Support_-_Certificate

So, any objections to adding the feature back to those branches? 

Also, the documentation should be fixed on https://grails.org/plugin/tomcat, to let users know the feature is not currently supported for:
grails.tomcat.certificateFile - The path to the OpenSSL certificate file
grails.tomcat.certificateKeyFile - The path to the OpenSSL certificate private key file
grails.tomcat.certificateKeyPassword - The password for the OpenSSL certificate private key file

**Comment by burtbeckwith** [comment]
fixed in 8.0.33 and 9.0.0.M4.1
